builder/relay.go

```diff
@@ -151,12 +160,35 @@ func (r *RemoteRelay) SubmitBlock(msg *boostTypes.BuilderSubmitBlockRequest, _ V
 
 func (r *RemoteRelay) SubmitBlockCapella(msg *capella.SubmitBlockRequest, _ ValidatorData) error {
 	log.Info("submitting block to remote relay", "endpoint", r.endpoint)
-	code, err := server.SendHTTPRequest(context.TODO(), *http.DefaultClient, http.MethodPost, r.endpoint+"/relay/v1/builder/blocks", msg, nil)
-	if err != nil {
-		return fmt.Errorf("error sending http request to relay %s. err: %w", r.endpoint, err)
+
+	endpoint := r.endpoint + "/relay/v1/builder/blocks"
+	if r.cancellationsEnabled {
+		endpoint = endpoint + "?cancellations=true"
 	}
-	if code > 299 {
```

Comment:
This is not your change but I wonder why we use 299 instead of `!= http.StatusOk`

Reply:
had that for a long time, because any 2xx error is okay, and starting with 300 is an error

Reply:
`because any 2xx error is okay`

I don't know if this is true. There are certain 2XX responses that yield partial or no content. I don't think it is good practice to handle such events as if they were valid even if the request itself was "successful" 

Such conditional also does not check for 1XX responses. It would be better in our application layer to handle these scenarios with more granularity otherwise it can be very difficult to debug and yield misleading results

Reply:
makes sense. let's revisit this another time though as it's outside the scope of this PR